### DISCUSSION TITLE
Grandorgueperftest

### DISCUSSION THREAD

**Discussion by paolo2504** [GrandOrgue/grandorgue#1089] [Developers]
Please, what exactly does Grandorgueperftest now present in latest release ? I tried launching it with -h option and nothing seems to happen.
Thanks.

**Comment by oleg68** [comment]
1. GrandOrguePerfTest is a small performance-testing tool of some GrandOrgue sound engine features and settings. I think @larspalo could tell more.
2. Unfortunally GrandOrguePerfTest.exe now has the same problems that GrandOrgueTool.exe had earlier: Windows does not recognize it as a console application, so it prints nothing. Fixing it requires a huge volume of work.

The workaround is to run GrandOrguePerfTest.exe with the output redirection

```
GrandOrguePerfTest.exe >perftest.log
```

This application will run in 10-15 minutes in background and it will fill perftest.log with it's result.

**Reply by larspalo** [reply]
The workaround is absolutely sufficient for now, I think. We just need to document it somehow so that windows users wanting to test it will know about it and not expect something else. Fixing so that windows understands that it's a console app is definitely not a high priority compared to the GrandOrgueTool (which has greater usage value) in my opinion.

**Reply by paolo2504** [reply]
Ok thanks you all. 
Im just curious… i’ll give a try with redirection.

**Reply by paolo2504** [reply]
i made a quick test. Apparently the tool does nothing. It uses near 80%cpu and 25% memory, the redirected log file is open but after 5-6 minutes stays at 0 bytes. After 10 minutes i had to kill process.
The config file is present under roaming folder but it looks like settings by default ones as they dont correspond to GO settings.
For example i set memory limit 0 and here i see 7897, i set loseless compression and here i see =N....
Just to inform you. Regards.

........
.........
InterpolationType=0
Language=
LoadConcurrency=1
LoadLastFile=Y
LoopLoad=2
LosslessCompression=N
MIDIPlayerPath=
MIDIRecorderPath=
ManageCache=Y
ManagePolyphony=Y
MemoryLimit=7897.000000
OrganCount=0
OrganPackagePath=
OrganPath=

.......................

**Reply by larspalo** [reply]
You need more patience! It takes quite a while to complete all the tests.

What the perftest does is to stress test the system for many different settings, including different sample bit sizes and samplerates as well as types of interpolation. The results are how much polyphony is available under optimal circumstances, in practical terms available polyphony will most likely be slightly less inside GO than what the test tool reports.

The main reason for using it is actually mostly for developers to test how any changes to the sound engine will affect the available polyphony in GO. For instance when I experimented with the filter version of swell box modeling I used it a lot. For normal users one single run on a particular system will be enough to determine what kind of expectations on polyphony one can have. This can of course be determined by trial and error too, which for users might actually be better since the test app goes on to the point where output will be totally overloaded... In playing one could experience troubles much earlier than at max polyphony.

**Reply by paolo2504** [reply]
Really appreciate your detailed explanation. 
best regards
